contracts/VaultV3.vy

```diff
@@ -1276,15 +1315,21 @@ def _process_report(strategy: address) -> (uint256, uint256):
     if gain > 0:
         # NOTE: this will increase total_assets
         current_debt = unsafe_add(current_debt, gain)
-        self.strategies[strategy].current_debt = current_debt
-        self.total_debt += gain
-
+        if strategy != self:
+            self.strategies[strategy].current_debt = current_debt
+            self.total_debt += gain
+        else:
+            self.total_idle += gain
+        
     # Or record any reported loss
     elif loss > 0:
         current_debt = unsafe_sub(current_debt, loss)
-        self.strategies[strategy].current_debt = current_debt
-        self.total_debt -= loss
-
+        if strategy != self:
+            self.strategies[strategy].current_debt = current_debt
+            self.total_debt -= loss
+        else:
+            self.total_idle -= loss
```

Comment:
It should not be no. Since we always update total_idle based on the actual amounts even in the case of rounding problems

But seems we might as well add it in just in case.